tools/platformio-build.py

```diff
@@ -258,17 +259,27 @@ def scons_patched_match_splitext(path, suffixes=None):
     env.Append(CPPDEFINES=[current_vtables])
 assert current_vtables
 
+current_mmu_iram_size = None
+for d in flatten_cppdefines:
+    if str(d).startswith("MMU_IRAM_SIZE"):
+        current_mmu_iram_size = d
```

Comment:
```suggestion
mmu_iter = iter(flatten_cppdefines)
while True:
    try:
        item = next(mmu_iter)
        if str(item).startswith("MMU_IRAM_SIZE"):
            val = next(mmu_iter)
            current_mmu_iram_size = "{}={}".format(item, val)
    except StopIteration:
        break
```
This is a weird one. flattened list value is actually right after the d match, so we never get the real flag into the build. I just noticed this when building, iram len was 1. Directly iterating with env["CPPDEFINES"] is weird, too, since we either get a 2-item tuple or a string. SCons leaves this as an exercise for the user :)
sry not noticing this earlier.

Reply:
Or, directly in defines:
```suggestion
for flag in env["CPPDEFINES"]:
    try:
        d, val = flag
        if str(d).startswith("MMU_IRAM_SIZE"):
            current_mmu_iram_size = "{}={}".format(d, val)
    except ValueError:
        continue
```